internal/filter/filter.go

```diff
@@ -7,32 +7,38 @@ import (
 
 var matchGroups = regexp.MustCompile(`(?:([^\s"]+)|"([^"]*)"?)+`)
 
-// Split terms by spaces, except for spaces within quotes.
-// Like http://stackoverflow.com/questions/16261635/javascript-split-string-by-space-but-ignore-space-in-quotes-notice-not-to-spli
-// except:
-// - Treat the query ["san fran] as having the single term ["san fran"] instead of ["san", "fran"]
-// - Removes the delimiting quotes.
-func Match(str, query string) bool {
+func GetMatchStrings(query string) []string {
 	if query == "" {
-		return true
+		return nil
 	}
-
 	matches := matchGroups.FindAllStringSubmatch(query, -1)
-	for _, match := range matches {
+	matchStrings := make([]string, len(matches), len(matches))
```

Comment:
woah, this should not be preallocated.

We want 
```
matchStrings := make([]string, 0, len(matches))
for _, match := range matches {
 ...
  matchStrings = append(matchStrings, matchString)
```

This is probably why we're seeing nils everywhere.